File: leetcode/array/intersection-of-two-arrays-ii.cpp
class Solution {
public:
  vector<int> intersect(vector<int> &nums1, vector<int> &nums2) {
    int r[1001] = {0};
    for (auto c : nums1)
      r[c]++;

    vector<int> s;
    for (auto c : nums2) {
      if (r[c] > 0) {
        s.push_back(c);
        r[c]--;
      }
    }
    return s;
  }
};

File: leetcode/array/intersection-of-two-arrays-ii.go
func intersect(nums1 []int, nums2 []int) []int {
	var record [1001]int
	for _, v := range nums1 {
		record[v]++
	}

	var results []int
	for _, v := range nums2 {
		if record[v] != 0 {
			results = append(results, v)
			record[v]--
		}
	}

	return results
}
